File: test/csharp/dotnetsdk/http-service/success-runtime-config/Configuration/FixtureConfiguration.cs
using System.Text.RegularExpressions;
namespace Fixture;
public sealed record FixtureConfiguration(int Port, string Mode, int Status, string Label)
{
    public static FixtureConfiguration Load()
    {
        var raw = Environment.GetEnvironmentVariable("PORT") ?? "";
        if (!Regex.IsMatch(raw, "^[0-9]{1,5}$") || !int.TryParse(raw, out var port) || port < 1 || port > 65535)
            throw new ArgumentException("Invalid PORT");
        var mode = "config";
        var label = mode == "config" ? Environment.GetEnvironmentVariable("FIXTURE_LABEL") ?? "runtime-config-default" : "deployment-smoke-ok";
        return new FixtureConfiguration(port, mode, 200, label);
    }
}

File: test/single-language/java/jdk/http-service/failure-health-rollback/src/acceptance/config/FixtureConfiguration.java
package acceptance.config;

public record FixtureConfiguration(int port, String label) {
    public static FixtureConfiguration load() {
        String raw = System.getenv("PORT");
        if (raw == null || !raw.matches("[0-9]{1,5}"))
            throw new IllegalArgumentException("Invalid PORT");
        int port = Integer.parseInt(raw);
        if (port < 1 || port > 65535)
            throw new IllegalArgumentException("Invalid PORT");
        String label = System.getenv("FIXTURE_LABEL");
        return new FixtureConfiguration(port,
                mode().equals("config") ? (label == null ? "runtime-config-default" : label) : "deployment-smoke-ok");
    }

    public static String mode() {
        return "smoke";
    }

    public static int status() {
        return 503;
    }
}
